packages/runner/src/run.ts

```diff
@@ -105,6 +105,14 @@ async function callCleanupHooks(cleanups: HookCleanupCallback[]) {
 }
 
 export async function runTest(test: Test, runner: VitestRunner) {
+  // @ts-expect-error untyped global
```

Comment:
runner cannot depend on private vitest properties, everything should be provided with the runner class

Reply:
I don't know the best solution. We can also add a method there specifically for canceling, I suppose.

Reply:
Alright I'm planning a `Runner.onCancel` method that runners can use to decide how they should cancel the runs. This seems to work nice for `worker_threads` and `child_process` pools but I'll need to figure out browser next. 

```ts
class CustomRunner {
  private cancelRun = false;

  /**
   * Called when test runner should cancel next test runs.
   * Runner should listen for this method and mark tests and suites as skipped in
   * "onBeforeRunSuite" and "onBeforeRunTest" when called.
   */
  onCancel() {
    this.cancelRun = true;
  }

  async onBeforeRunTest(test: Test) {
    if (this.cancelRun) test.mode = "skip";
    // ...
  }

  onBeforeRunSuite(suite: Suite) {
    if (this.cancelRun)
      suite.mode = 'skip'
    // ...
  }
}
```